### PR TITLE
Several changes to make deployment easier.

### PR DESCRIPTION
**Pull request by manthey**
The swagger api pages would only work if the static_root was absolute, because this was otherwise treated as relative to the api path.  I've added a new api_static_root parameter, which now allows all of the paths to be relative.  This makes it easier, for instance, for proxies to get to these locations.

The api_root is placed in the swagger page as well, as this also also the sub-describe calls to use a relative path instead of the entire query path.

Fixed the swagger throbber using css to override its path.  This still throws a 404 in the browser's console, when the original location is requested, but the throbber is properly displayed.  This fixes issue #446.

Add documentation that you don't need to specify a tools.proxy.base or tools.proxy.local if the proxy server adds X-Forwarded-Host.

Expand the user path element when parsing the log root, not just in the constants.  This allows a conf file to use ~ in the log root.

### CODE REVIEW

File: clients/web/static/girder-swagger.js [hunk 1]
```diff
@@ -1,6 +1,11 @@
 $(function () {
+    var apiRoot = $('#g-global-info-apiroot').text().replace(
+        '%HOST%', 'http://' + window.location.host);
```
Comment (cpatrick) [inline clients/web/static/girder-swagger.js]: What happens if we're https only?

Comment (manthey) [inline clients/web/static/girder-swagger.js]: This replace only occurs in tests, not in actual deployment.  We don't
currently test the swagger docs (but we should -- see issue #320), so this
doesn't matter yet.

On Fri, Apr 3, 2015 at 10:25 AM, Patrick Reynolds <notifications@github.com>
wrote:

> In clients/web/static/girder-swagger.js
> <https://github.com/girder/girder/pull/788#discussion_r27731665>:
>
> > @@ -1,6 +1,11 @@
> >  $(function () {
> > +    var apiRoot = $('#g-global-info-apiroot').text().replace(
> > +        '%HOST%', 'http://' + window.location.host);
>
> What happens if we're https only?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/girder/girder/pull/788/files#r27731665>.
>



-- 
David Manthey
R&D Engineer
Kitware Inc.
(518) 881-4439

Comment (ronichoudhury) [inline clients/web/static/girder-swagger.js]: I assume "window.location.protocol" could replace the hard-coded http://?

roni

Comment (manthey) [inline clients/web/static/girder-swagger.js]: That would be nicer (this same replace is also in clients/web/src/init.js).

Comment (zachmullen) [inline clients/web/static/girder-swagger.js]: Doesn't `window.location.origin` capture the protocol, host, and port?

Comment (manthey) [inline clients/web/static/girder-swagger.js]: Does IE support `window.location.origin` yet?  Do we care?

Comment (manthey) [inline clients/web/static/girder-swagger.js]: In answer to my last question, IE 11 supports it, and we don't care.

File: girder/api/describe.py [hunk 2]
```diff
@@ -139,76 +139,83 @@ class ApiDocs(object):
...
+.response_throbber {
+  content:url("${staticRoot}/built/swagger/images/throbber.gif");
+}
+    </style>
+  </head>
+  <body>
+    <div class="docs-header">
+      <span>Girder REST API Documentation</span>
+      <i class="icon-book-alt right"></i>
+      <div id="g-global-info-apiroot" class="hide">${apiRoot}</div>
```
Comment (zachmullen) [inline girder/api/describe.py]: This doesn't appear to be hidden correctly, maybe need to just directly set `display: none`.

screen shot 2015-04-05 at 8 04 06 pm

**Comment by zachmullen** [comment]
LGTM

**Pull request merged**